schema_automator/importers/xsd_import_engine.py

```diff
@@ -0,0 +1,611 @@
+        for child in el:
+            if child.tag == f"{{{XSD}}}extension":
+                self.visit_complex_content_child(child, cls)
+            elif child.tag == f"{{{XSD}}}restriction":
+                self.visit_complex_content_child(child, cls)
+            elif child.tag == f"{{{XSD}}}annotation":
+                cls.description = self.visit_annotation(child, cls.description)
+                
+    def visit_complex_type(self, el: etree._Element, cls: ClassDefinition) -> None:
+        """
+        Adds information from a `<xsd:complexType>` onto a ClassDefinition
+
+        See 3.4 Complex Type Definitions: https://www.w3.org/TR/xmlschema11-1/#Complex_Type_Definitions
+        """
+        # Update the class name using the complex type's name, if we don't already have one
+        if cls.name == PLACEHOLDER_NAME and "name" in el.attrib:
+            cls.name = assert_type(el.attrib["name"], str)
+            cls.class_uri = urljoin(self.target_ns, cls.name) if self.target_ns else None
+
+        found_content = False
+        for child in el:
+            if child.tag == f"{{{XSD}}}complexContent":
+                self.visit_complex_content(child, cls)
+                found_content = True
+            elif child.tag == f"{{{XSD}}}simpleContent":
+                self.visit_simple_content(child, cls)
+                found_content = True
+
+        if not found_content:
+            """ 
+            3.4.2.3.2 Mapping Rules for Complex Types with Implicit Complex Content:
+            > When the <complexType> source declaration has neither <simpleContent> nor <complexContent> as a child, it is taken as shorthand for complex content restricting ·xs:anyType·. 
+            """
+            for child in el:
+                self.visit_complex_content_child(child, cls)
+
+    def visit_choice(self, el: etree._Element) -> Iterable[SlotDefinition]:
+        """
+        Converts an xsd:choice into a list of SlotDefinitions
+        """
+        # TODO: indicate that not all slots can be used at the same time
+        return self.visit_sequence(el)
+
+    def visit_sequence(self, el: etree._Element) -> Iterable[SlotDefinition]:
+        """
+        Converts an xsd:sequence into a list of SlotDefinitions
+        """
+        for child in el:
+            if child.tag == f"{{{XSD}}}element":
+                yield self.visit_element(child)
+            elif child.tag == f"{{{XSD}}}choice":
+                yield from self.visit_choice(child)
+            elif child.tag == f"{{{XSD}}}sequence":
+                yield from self.visit_sequence(child)
+            elif isinstance(child, etree._Comment):
+                pass
+            else:
+                print(f"Skipping unknown tag {child.tag}")
+
+    def visit_schema(self, schema: etree._Element) -> None:
+        """
+        Converts an xsd:schema into a SchemaDefinition
+        """
+        self.target_ns = schema.attrib.get("targetNamespace")
```

Comment:
Setting target_ns should have a default or error handling of some sort.
In the example I came up with, this ended up being None, which then caused the uri to be none at a later point.